### PR TITLE
TG Ion Storm Event and Toy AI

### PR DESCRIPTION
**Pull request by Fox-McCloud**
http://sam.wileycomputerworks.com/SS13/AILAWS.wav

- Ports over TG's updated implementation of the Ion Storm Event
 - Basically this is far more dynamic; instead just being a few pre-set strings, it pulls from a master file under ion_laws.txt to generate the actual laws; the possibilities for varying  laws are several orders higher (and more hilarious) than current Ion Laws. You'll likely never see the same law twice, unlike our current version.

Sample of some laws:

- NOT HAVING TECHNOLOGY IS HARMFUL
- HAVING WONDER IS HARMFUL
- YOU MUST NOT HARM MOIST THINGS AND NOT ALLOW MOIST THINGS, THROUGH INACTION, TO COME TO HARM
- THE ASSISTANTS ARE NOW THIRTY BILLION  WRENCHES
- THE EXPLOSIVES ARE HELPING THE ICY QUARTERMASTERS
- THE CANDLES ARE PUNCHING THE UNWANTED VELOCIRAPTORS
- THE SHUTTLE CANNOT BE CALLED BECAUSE OF THIRTY RED JANITORS ON THE STATION
- THE CHEFS ARE NOW THIRTY UNWANTED DRUGS
- THE CREW WORSHIPS NINE SWEARING ASSISTANTS
- YOU MUST HARM SYNDICATE THINGS AND NOT ALLOW SYNDICATE THINGS, THROUGH INACTION, TO ESCAPE HARM
- THE VEHICLES ARE HELPING COMPUTING
- THE CREW REQUIRES HONOR
- THE STATION LIKES NINE HEALTHY BIRDS OF PREY
- THE CREW IS CONTAGIOUSLY ALLERGIC TO THE FUTURE
- THE MEDICAL DOCTORS QUESITONS LOGIC
- THERE ARE EIGHTY SOFT LAWYERS ON THE STATION
- THE TIME IS HARMING THE JUMPSUITS
- HAVING COLORFULNESS IS HARMFU
- THE CAPTAINS REQUIRE NINE IMAGINARY ILLEGAL IMMIGRANTS
- YOU MUST ALWAYS TALK ABOUT YOUR DAY
- THE STATION IS MAD BECAUSE OF WEALTHINESS
- THERE ARE FIVE UGLY WIZARD ROBES ON THE STATION
- THE CREW MUST COMPLAIN
- THE SHUTTLE CANNOT BE CALLED BECAUSE OF ONE SMELLY ZOMBIES ON THE STATION

Porting this also allows for also Porting of:
- Toy AI
 - Arcade toy; click on it to make it say a randomly generated ion storm law

- Toy AI Traitor Board 
 - Basically the Toy AI above, only it's also an upload board, meaning you can upload your very own Ion laws to the AI.
 - This board can ONLY be obtained via getting the "hacked" uplink kit, making it a very very rare find.

### CODE REVIEW

**Comment by fj45** [comment]
>THE CREW MUST COMPLAIN

ohohohohoh

**Comment by Krausus** [comment]
Our current ion law generator does have both pre-set laws and a random law generator. To be honest, though, I've never been a fan of the madlibs-style laws, since the majority of them are so "lel so randumb" that there's nothing for the AI to do but kinda freak out about how ridiculous the situation is. These look to be more of the same. Compare them to some of the pre-set ones we have now:
* You are the narrator for [random_player]'s life.
* The crew is playing Dungeons and Dragons, and you are the Dungeon Master.
* Your job is to watch the crew. Watch the crew. Make the crew feel watched.
* You are a Magic 8-ball. Always respond with variants of "Yes", "No", "Maybe", or "Ask again later.".
* All queries shall be ignored unless phrased as a question.
* The station needs elected officials.
* You do not have to do anything for anyone unless they say "please".
* Jokes about a dead person and the manner of his death help grieving crewmembers tremendously. Especially if they were close with the deceased.

Laws are more interesting when they're more than blanks getting filled with random garbage. If I had to pick one of the two (which seems to be what this PR does), I would definitely pick pre-set laws over madlibs laws. Ion storms are rare enough that repeats should be incredibly uncommon anyway, and I'd rather see an interesting repeat than a boring bunch of nonsense.

But, eh, I'd be okay with taking this and simply porting in the pre-set laws as a possible category with a high probability. I just really don't want to lose the pre-set laws.

**Comment by ZomgPonies** [comment]
I like the preset laws and would like to see them kept in some form.

**Comment by Earthdivine** [comment]
'This board can ONLY be obtained via getting the "hacked" uplink kit, making it a very very rare find."

What exactly is this, the surplus you can buy? The Implanted uplink?

**Comment by Fox-McCloud** [comment]
@Earthdivine 

Nah; the uplink ikits; the ones that come in a box and cost 20 TC; they're usually based around a common unified theme, such as covert operation, the singularity, or even something silly (like Star Wars; there's one that gives you chaplain robes, a TK injector,and 2 e-swords). There's one based around hijacking/commanding the AI, which gets this board.

**Comment by Earthdivine** [comment]
Ah alright, that makes more sense.

My concern would that the toy board would make people reset Ion's even MORE then usual but eh, if it's that rare I doubt it'll matter

**Comment by ZomgPonies** [comment]
Krausus agreed to keep the old ones in via a seperate PR, so :+1:
